### ISSUE TITLE
Allow const ctors

### ISSUE THREAD

**Issue by tjammer** [tjammer/schmu#23]
I've disabled them for now because I couldn't make them work within a short time. Will revisit later once variants and pattern matching are more stable.
In theory though, they are similar enough to records that all the code can be adapted from there

**Comment by tjammer** [comment]
Const records might be broken as well.
See
```
(type t {:i& int :a int :b int})
(def const {:i 10 :a 10 :b 10})
```

**Comment by tjammer** [comment]
Fixed in 8d71de2d29bba549c67f50e936e5db7100359a3d
